src/main/java/com/thealgorithms/maths/PrimeCheck.java

```diff
@@ -82,4 +88,27 @@ private static long modPow(long a, long b, long c) {
         }
         return res % c;
     }
+
+    /**
+     * Checks if a given number is prime using an optimized approach.
+     * @param number the number to check
+     * @return true if the number is prime, false otherwise
+     */
+    public static boolean isPrimeNumberOptimised(long number) {
+        // Numbers less than or equal to 1 are not prime
```

Comment:
What do you think about removing these comments and moving them to the Javadoc? Because the code of this method is quite simple, but all these descriptions are harder to read: and after deleting it should be:

```
/**
 * Determines if a given number is prime using an optimized approach.
 * <p>
 * This method checks for primality with the following steps:
 * <ul>
 *   <li>Numbers less than or equal to 1 are not prime.</li>
 *   <li>Numbers 2 and 3 are prime.</li>
 *   <li>Eliminates even numbers and multiples of 3 early for efficiency.</li>
 *   <li>Uses the 6k ± 1 optimization for checking factors, which skips even numbers
 *       and multiples of 3 to cover all other possible factors.</li>
 *   <li>If no factors are found up to the square root of the number, the number is prime.</li>
 * </ul>
 * 
 * @param number the number to check
 * @return true if the number is prime, false otherwise
 */
public static boolean isPrimeNumberOptimised(long number) {
    if (number <= 1) return false;
    if (number <= 3) return true;
    if (number % 2 == 0 || number % 3 == 0) return false;
    for (long i = 5; i * i <= number; i += 6) {
        if (number % i == 0 || number % (i + 2) == 0) {
            return false;
        }
    }
    return true;
}
```